lib/spec_store.rb

```diff
@@ -253,69 +289,91 @@ def get_id_lists(init_diagnostics = nil)
 
       begin
         server_id_lists = JSON.parse(response)
-        local_id_lists = @specs[:id_lists]
-        if !server_id_lists.is_a?(Hash) || !local_id_lists.is_a?(Hash)
-          return
-        end
-        tasks = []
-
-        if server_id_lists.length == 0
-          return
-        end
+        process_id_lists(server_id_lists, init_diagnostics)
+      rescue
+        # Ignored, will try again
+      end
+    end
 
-        init_diagnostics&.mark("get_id_lists", "start", "process", server_id_lists.length)
+    def process_id_lists(new_id_lists, init_diagnostics, from_adapter: false)
+      local_id_lists = @specs[:id_lists]
+      if !new_id_lists.is_a?(Hash) || !local_id_lists.is_a?(Hash)
+        return
+      end
+      tasks = []
 
-        server_id_lists.each do |list_name, list|
-          server_list = IDList.new(list)
-          local_list = get_id_list(list_name)
+      if new_id_lists.length == 0
+        return
+      end
```

Comment:
what if the old list had ids?  Need to handle the removal/deletion case

Reply:
What do you mean? If you select "hide whitespace" in diff view, you'll see this function logic wasn't changed. It's just showing it weird cus the line numbers don't match up